docs/spec/index.bs

```diff
@@ -59,6 +68,63 @@ Markup Shorthands: markdown yes, css no
 
 <i>Todo</i>
 
+## {{AttesterConnection}} ## {#attester-connection}
+
+<xmp class="idl">
+  interface AttesterConnection {
+    ArrayBuffer getAttestation(DOMString contentBinding);
+  };
+</xmp>
+
+: getAttestation
+:: Returns a COSE signed CBOR object as an ArrayBuffer from the [=attester=] that contains the [=Integrity verdict=].
+
+# Web Environment Integrity API # {#api}
+
+## Extensions to {{Navigator}} ## {#extensions-to-navigator}
+
+<xmp class="idl">
+  [Exposed=Window]
+  partial interface Navigator {
+    [SecureContext] Promise<EnvironmentIntegrity> getEnvironmentIntegrity(DOMString contentBinding);
+  };
+</xmp>
+
+### {{Navigator/getEnvironmentIntegrity()}} ### {#navigator-getenvironmentintegrity}
+
+<div algorithm="navigator-getenvironmentintegrity-alg">
+    The [=user agent=] has the global <dfn for="attesterConnection">attesterConnection</dfn>, which is
+    an {{AttesterConnection}} with the [=attester=].
+
+    The <dfn method for="Navigator"><code>getEnvironmentIntegrity(|contentBinding|)</code></dfn> method, when invoked, runs these steps:
+
+    1. Let |promise| be [=a new promise=]
+    1. Run the following steps [=in parallel=]:
+        1. Let |environmentIntegrity| be a new {{EnvironmentIntegrity}}
+        1. Set |environmentIntegrity|.{{EnvironmentIntegrity/attestationToken}} to [=attesterConnection=].<a method for=AttesterConnection>getAttestation(|contentBinding|)</a>. If this fails then:
+            1. [=Reject=] |promise| with a <i>TODO</i> [=Exception=]
+            1. Abort these steps
+        1. [=Resolve=] |promise| with |environmentIntegrity|
+    1. Return |promise|
+</div>
+
+## {{EnvironmentIntegrity}} ## {#environment-integrity}
+
+<xmp class="idl">
+  interface EnvironmentIntegrity {
+    readonly attribute ArrayBuffer attestationToken;
+    DOMString encode();
+    [Default] object toJSON();
+  };
+</xmp>
+
+: attestationToken
+:: The attestation token is a COSE signed CBOR object as an ArrayBuffer from the attester.
+: encode()
+:: The encode method will return a Base64 string representation of the attestation token.
+: toJSON()
+:: The toJSON method returns a JSON representation of the attestation token. Useful for local debugging.
```

Comment:
i'd mention that this is human readable after the CBOR payload has been decoded

Reply:
Makes sense!